src/Composer/Action/UpdatePackage.php

```diff
@@ -32,17 +31,22 @@ public function __construct(Application $app)
      *
      * @param  $packages array Indexed array of package names to update
      * @param  $options  array [Optional] changed option set
-     * @return integer 0 on success or a positive error code on failure
+     *
+     * @return int 0 on success or a positive error code on failure
+     * @throws \Bolt\Exception\PackageManagerException
      */
-    public function execute(array $packages = array(), array $options = null)
+    public function execute(array $packages = array(), array $options = array())
     {
+        /** @var $composer \Composer\Composer */
         $composer = $this->app['extend.manager']->getComposer();
         $io = $this->app['extend.manager']->getIO();
-        $options = $this->app['extend.manager']->getOptions();
+        $packageManagerOptions = $this->app['extend.manager']->getOptions();
 
         // Handle passed in options
-        if (!is_null($options)) {
-            $options = Arr::mergeRecursiveDistinct($options, $options);
+        if (!$options) {
+            $options = array_replace_recursive($packageManagerOptions, $options);
```

Comment:
Pseudo example:
```php
$a = array(
    'foo' => array(
        'cow' => 'moo',
        'dog' => 'whoof'
    )
);

$b = array(
    'foo' => array(
        'cow' => 'steak'
    )
);
```

Merging those would wipe the 'dog' key.  You might need to go one deeper, can't remember.